File: repeatedly-until/app/src/main/java/com/sqisland/espresso/repeatedly_until/MainActivity.java
package com.sqisland.espresso.repeatedly_until;


import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends Activity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    final TextView numberView = findViewById(R.id.number);
    final TextView squaredView = findViewById(R.id.squared);

    numberView.setText("0");
    squaredView.setText("0");

    numberView.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View view) {
        int number = Integer.parseInt(numberView.getText().toString());
        number += 1;
        int squared = number * number;
        numberView.setText(String.valueOf(number));
        squaredView.setText(String.valueOf(squared));
      }
    });
  }
}

File: repeatedly-until/app/src/main/java/com/sqisland/espresso/repeatedly_until/MainActivity.kt
package com.sqisland.espresso.repeatedly_until


import android.app.Activity
import android.os.Bundle
import android.widget.TextView

class MainActivity : Activity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    val numberView = findViewById<TextView>(R.id.number)
    val squaredView = findViewById<TextView>(R.id.squared)

    numberView.text = "0"
    squaredView.text = "0"

    numberView.setOnClickListener {
      var number = Integer.parseInt(numberView.getText().toString())
      number += 1
      val squared = number * number
      numberView.text = number.toString()
      squaredView.text = squared.toString()
    }
  }
}